esp-hal/src/lib.rs

```diff
@@ -572,3 +572,47 @@ pub fn init(config: Config) -> Peripherals {
 
     peripherals
 }
+
+/// Asynchronous utilities.
+pub mod asynch {
```

Comment:
Nitpick: I'm not sure we need the mod either, but I'm not too fussed, so feel free to ignore